metrics/solana.py

```
@@ -113,6 +114,79 @@ def get_params_from_state(state_data: dict) -> list:
+        """Build getAccountInfo params pinned to the settled slot."""
+        anchor_slot = int(state_data["agreement_anchor"])
+        return [
+            _AGREEMENT_PROBE_ACCOUNT,
+            {
+                "encoding": "base64",
+                "commitment": "finalized",
+                "minContextSlot": anchor_slot,
+            },
+        ]
+
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
```

Ruff (0.15.13)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `*args`

(ANN401)